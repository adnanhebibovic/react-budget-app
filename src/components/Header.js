import React from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'

import {startLogout} from '../actions/auth'

function Header(props) {
    if (!props.isAuthenticated) {
        return null;   
    }

    return (
        <header>
            <h1>Budget app</h1>
            <ul>
                <li>
                    <NavLink to="/dashboard">Dashboard</NavLink>
                </li>
                <li>
                    <NavLink to="/expenses">Create</NavLink>
                </li>
            </ul>
            <button onClick={props.startLogout}>Logout</button>
        </header>
    )
}

const mapStateToProps = (state) => {
    return {
        isAuthenticated: !!state.auth.uid
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        startLogout: () => dispatch(startLogout())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Header)