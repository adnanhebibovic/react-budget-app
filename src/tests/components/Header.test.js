import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { create, act } from 'react-test-renderer'
import Header from '../../components/Header'

test('Should create Header component to match snapshot', () => {
    let header
    act(() => {
        header = create(<BrowserRouter><Header></Header></BrowserRouter>)
    })
    expect(header.toJSON()).toMatchSnapshot()
})
