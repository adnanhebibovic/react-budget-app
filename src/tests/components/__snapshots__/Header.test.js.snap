// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should create Header component to match snapshot 1`] = `
<header>
  <h1>
    Budget app
  </h1>
  <ul>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Dashboard
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/expenses"
        onClick={[Function]}
      >
        Create
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/help"
        onClick={[Function]}
      >
        Help
      </a>
    </li>
  </ul>
</header>
`;
